### PR TITLE
sasarkar/fusedrope inp bf16

### PR DESCRIPTION
**Pull request by ssarkar2**
# What does this PR do?
pulling in this: https://github.com/HabanaAI/optimum-habana-fork/commit/1d444339b146c3bc000b23b53fdf256232d75964

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by regisss — COMMENTED** [review]
Have you checked the accuracy of the trained models with this change? RoPE is quite sensitive to the data type that is used to compute it.

**Comment by ssarkar2** [comment]
Tried a small text gen expt:

```
python run_generation.py --model_name_or_path /mnt/weka/data/llama_inference/Llama-2-7b-chat-hf/ --use_kv_cache --max_new_tokens 100 --bf16 --batch_size 4  --use_hpu_graph --trim_logits --bucket_size 128 --bucket_internal --reuse_cache --use_flash_attention
```

```
 python run_generation.py --model_name_or_path /mnt/weka/data/llama_inference/Llama-2-7b-chat-hf/ --use_kv_cache --max_new_tokens 1024 --max_input_tokens 1024 --bf16 --batch_size 4  --use_hpu_graph --trim_logits --bucket_size 128 --bucket_internal --reuse_cache --use_flash_attention
```

Got exactly same outputs


Let me check for the training results

**Review by regisss — APPROVED** [review]
